content/cloud-docs/registry/publish-providers.mdx

```diff
@@ -41,6 +41,8 @@ To publish a new provider through the Terraform Cloud API:
     ```json
     "key-id": "32966F3FB5AC1129"
     ```
+1. If the provider does not yet exist in the private registry use the [Create a Provider endpoint](/https://www.terraform.io/cloud-docs/api-docs/private-registry/providers#create-a-provider) to create it. If you are publishing a new version for an already existing provider skip this step.
```

Comment:
Thank you!  I'm wondering - could we put this as the first step in the process before the GPG key endpoint step? I think that would help everything flow better and would make logical sense :-) 

```suggestion
1. Use the [Create a Provider endpoint](/https://www.terraform.io/cloud-docs/api-docs/private-registry/providers#create-a-provider) to create the provider in Terraform Cloud. 
```

Reply:
Telling users that they can skip the provider creation does indeed seem a bit redundant :) 

To me, personally having the step to create the GPG key first seems more logical as the GPG key is something that can and likely will be used across multiple providers in this namespace. Then we have a sequence of tightly related resources - Provider > Provider versions > Provider version platforms.

Still, I would ultimately defer to you on this so if you feel that it is better we can change it.

Reply:
okay! I trust your judgment - let's have this be the second step, just after the GPG key :-).